homeassistant/components/huawei_lte/device_tracker.py

```diff
@@ -1,63 +1,52 @@
 """Support for device tracking of Huawei LTE routers."""
 
 import logging
-from typing import Any, Dict, List, Optional
 
 import attr
-import voluptuous as vol
 
-import homeassistant.helpers.config_validation as cv
-from homeassistant.components.device_tracker import PLATFORM_SCHEMA, DeviceScanner
+from homeassistant.components.device_tracker import SOURCE_TYPE_ROUTER
 from homeassistant.const import CONF_URL
-from . import RouterData
+from homeassistant.helpers.event import track_time_interval
+from . import MIN_TIME_BETWEEN_UPDATES, RouterData
 from .const import DOMAIN, KEY_WLAN_HOST_LIST
 
 
 _LOGGER = logging.getLogger(__name__)
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({vol.Optional(CONF_URL): cv.url})
 
-HOSTS_PATH = f"{KEY_WLAN_HOST_LIST}.Hosts.Host"
+def setup_scanner(hass, config, see, discovery_info=None):
+    """Set up Huawei LTE device tracker scanner."""
+    if discovery_info is None:
+        return
 
-
-def get_scanner(hass, config):
-    """Get a Huawei LTE router scanner."""
-    data = hass.data[DOMAIN].get_data(config)
-    data.subscribe(HOSTS_PATH)
-    return HuaweiLteScanner(data)
+    router_data = hass.data[DOMAIN].get_router_data(discovery_info[CONF_URL])
+    scanner = HuaweiLteScanner(router_data, see)
+    scanner.update()
+    track_time_interval(hass, scanner.update, MIN_TIME_BETWEEN_UPDATES)
+    return True
 
 
 @attr.s
-class HuaweiLteScanner(DeviceScanner):
+class HuaweiLteScanner:
```

Comment:
We should implement the new `ScannerEntity` entity class.

https://github.com/home-assistant/home-assistant/blob/e0f1677296e6b7f06e393b4d341bc247de169d53/homeassistant/components/device_tracker/config_entry.py#L120

See unifi and zha device trackers for example.

Reply:
Isn't that limited to/supposed to be only used with config entries? See comment below for not wanting to limit to those.

Reply:
Yes. We have to use config entries. We'll import the config yaml into the config flow and get a config entry.